Undo the unintended parts of a much-too-broad "GNU"-to-"GNU Lesser"
replace-all used by the developers sometime between the release of
xbase 3.1.1 and 3.1.2.

A similar but incorrect patch (that also contains unrelated fixes) is
attached here:
https://sourceforge.net/p/xdb/patches/5/

The problem was also mentioned on the mailing list, where a different
incorrect solution was proposed:
https://sourceforge.net/p/xdb/discussion/131854/thread/18d7fb7c/
--- xbase64/xbase64.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xbase64.cpp	2019-10-01 21:28:33.000000000 -0500
@@ -38,7 +38,7 @@
 
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xbase64.h"
 #endif
 
--- xbase64/xbase64.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbase64.h	2019-10-01 21:28:32.000000000 -0500
@@ -42,7 +42,7 @@
 #ifndef __XB_XBASE_H__
 #define __XB_XBASE_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
@@ -60,7 +60,7 @@
 // ripped from wxWindows
 
 // _declspec works in BC++ 5 and later, as well as VC++
-#if defined(__VISUALC__) || defined(__BORLANDC__) || defined(__GNU LesserC__)
+#if defined(__VISUALC__) || defined(__BORLANDC__) || defined(__GNUC__)
 #  ifdef XBMAKINGDLL
 #    define XBDLLEXPORT __declspec( dllexport )
 #    define XBDLLEXPORT_DATA(type) __declspec( dllexport ) type
--- xbase64/xbcdx.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbcdx.h	2019-10-01 21:28:33.000000000 -0500
@@ -41,7 +41,7 @@
   xbLong leftSibling;
   xbLong rightSibling;
 }
-#ifdef __GNU LesserC__
+#ifdef __GNUC__
 	__attribute__((packed))
 #endif
 ;
@@ -50,7 +50,7 @@
 {
   char keys[500];
 }
-#ifdef __GNU LesserC__
+#ifdef __GNUC__
 	__attribute__((packed))
 #endif
 ;
@@ -67,7 +67,7 @@
   char byteCount;
   char keys[488];
 } 
-#ifdef __GNU LesserC__
+#ifdef __GNUC__
 	__attribute__((packed))
 #endif
 ;
--- xbase64/xbdate.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xbdate.cpp	2019-10-01 21:28:32.000000000 -0500
@@ -40,7 +40,7 @@
 
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xbdate.h"
 #endif
 
--- xbase64/xbdate.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbdate.h	2019-10-01 21:28:33.000000000 -0500
@@ -45,7 +45,7 @@
 #ifndef __XB_XBDATE_H__
 #define __XB_XBDATE_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
--- xbase64/xbdbf.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xbdbf.cpp	2019-10-01 21:28:33.000000000 -0500
@@ -39,7 +39,7 @@
 
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xbdbf.h"
 #endif
 
--- xbase64/xbdbf.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbdbf.h	2019-10-01 21:28:33.000000000 -0500
@@ -41,7 +41,7 @@
 #ifndef __XB_DBF_H__
 #define __XB_DBF_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
--- xbase64/xbexp.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xbexp.cpp	2019-10-01 21:28:32.000000000 -0500
@@ -38,7 +38,7 @@
 
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xbexp.h"
 #endif
 
--- xbase64/xbexp.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbexp.h	2019-10-01 21:28:32.000000000 -0500
@@ -42,7 +42,7 @@
 #ifndef __XB_EXP_H__
 #define __XB_EXP_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
--- xbase64/xbfile.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xbfile.cpp	2019-10-01 21:28:33.000000000 -0500
@@ -39,7 +39,7 @@
 
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xbfile.h"
 #endif
 
--- xbase64/xbfile.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbfile.h	2019-10-01 21:28:33.000000000 -0500
@@ -45,7 +45,7 @@
 #ifndef __XB_FILE_H__
 #define __XB_FILE_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
--- xbase64/xbfilter.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xbfilter.cpp	2019-10-01 21:28:32.000000000 -0500
@@ -38,7 +38,7 @@
 
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xbfilter.h"
 #endif
 
--- xbase64/xbfilter.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbfilter.h	2019-10-01 21:28:32.000000000 -0500
@@ -44,7 +44,7 @@
 #ifndef __XB_FILTER_H__
 #define __XB_FILTER_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
--- xbase64/xbindex.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xbindex.cpp	2019-10-01 21:28:32.000000000 -0500
@@ -37,7 +37,7 @@
        USA
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xbindex.h"
 #endif
 
--- xbase64/xbindex.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbindex.h	2019-10-01 21:28:32.000000000 -0500
@@ -43,7 +43,7 @@
 #ifndef __XB_INDEX_H__
 #define __XB_INDEX_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
--- xbase64/xblock.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xblock.cpp	2019-10-01 21:28:32.000000000 -0500
@@ -42,7 +42,7 @@
        USA
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xblock.h"
 #endif
 
--- xbase64/xblock.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xblock.h	2019-10-01 21:28:33.000000000 -0500
@@ -44,7 +44,7 @@
 #ifndef __XB_XBLOCK_H__
 #define __XB_XBLOCK_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
--- xbase64/xbndx.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xbndx.cpp	2019-10-01 21:28:33.000000000 -0500
@@ -36,7 +36,7 @@
 
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xbndx.h"
 #endif
 
--- xbase64/xbndx.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbndx.h	2019-10-01 21:28:32.000000000 -0500
@@ -41,7 +41,7 @@
 #ifndef __XB_NDX_H__
 #define __XB_NDX_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
--- xbase64/xbntx.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xbntx.cpp	2019-10-01 21:28:32.000000000 -0500
@@ -40,7 +40,7 @@
        USA
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xbntx.h"
 #endif
 
--- xbase64/xbntx.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbntx.h	2019-10-01 21:28:33.000000000 -0500
@@ -42,7 +42,7 @@
 #ifndef __XB_NTX_H__
 #define __XB_NTX_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
--- xbase64/xbstring.cpp	2006-07-17 11:54:42.000000000 -0500
+++ xbase64/xbstring.cpp	2019-10-01 21:28:32.000000000 -0500
@@ -38,7 +38,7 @@
 
 */
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
   #pragma implementation "xbstring.h"
 #endif
 
--- xbase64/xbstring.h	2006-07-17 11:54:50.000000000 -0500
+++ xbase64/xbstring.h	2019-10-01 21:28:33.000000000 -0500
@@ -41,7 +41,7 @@
 #ifndef __XBSTRING_H__
 #define __XBSTRING_H__
 
-#ifdef __GNU LesserG__
+#ifdef __GNUG__
 #pragma interface
 #endif
 
